src/controllers/__test__/eventController.test.js

```diff
@@ -1,12 +1,181 @@
-const { searchEvents } = require('../eventController');
+const supertest = require('supertest');
+const app = require('../../app');
+
+const request = supertest(app);
 const Event = require('../../models/Events');
+const User = require('../../models/Users');
 const Tag = require('../../models/Tags');
+const {
+  createEvent,
+  getAllEvents,
+  searchEvent,
+  getEvent,
+  updateEvent,
+  deleteEvent,
+  joinEvent,
+} = require('../eventController');
 
-jest.mock('../../models/Events'); // Mock the Event model
+jest.mock('../../models/Events');
+jest.mock('../../models/Users');
 jest.mock('../../models/Tags'); // Mock the Tag model
 
+let mockEvent = {
+  _id: 'event-id',
+  title: 'Test Event',
+  hostId: 'host-id',
+  participants: [],
+  capacity: 50,
+  save: jest.fn(),
+};
+
+const mockUser = {
+  _id: 'user-id',
+  name: 'John Doe',
+  email: 'john@example.com',
+  events: [],
+  save: jest.fn(),
+};
+
+const mockReq = {
+  params: { eventId: mockEvent.id },
+  user: { id: mockUser.id },
+};
+
+const mockRes = {
+  status: jest.fn().mockReturnThis(),
+  json: jest.fn(),
+};
+
 beforeEach(() => {
-  jest.clearAllMocks(); // Clear mock calls before each test
+  mockEvent = {
+    _id: 'event-id',
+    title: 'Test Event',
+    hostId: 'host-id',
+    participants: [],
+    capacity: 50,
+    save: jest.fn(),
+  };
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('joinEvent', () => {
+  it('should add user to event participants and add event to user events', async () => {
```

Comment:
You should also mock email sending functionality because every time the tests run, an email is actually sent from our Gmail account.